dpgen/generator/run.py

```diff
@@ -3022,7 +3063,35 @@ def run_fp (iter_index,
         backward_files = ['output']
         run_fp_inner(iter_index, jdata, mdata,  forward_files, backward_files, _qe_check_fin, log_file = 'output')
     elif fp_style == "abacus/scf":
-        forward_files = ["INPUT", "STRU", "KPT"] + fp_pp_files
+        fp_params = {}
+        if 'user_fp_params' in jdata.keys() :
+            fp_params = jdata['user_fp_params']
+        elif 'fp_incar' in jdata.keys():
+            fp_input_path = jdata['fp_incar']
+            assert(os.path.exists(fp_input_path))
+            fp_input_path = os.path.abspath(fp_input_path)
+            fp_params = get_abacus_input_parameters(fp_input_path)
+
+        forward_files = ["INPUT", "STRU"]
+        if 'kspacing' not in fp_params.keys():
+            if 'gamma_only' in fp_params.keys():
+                if type(fp_params["gamma_only"])==str:
+                    fp_params["gamma_only"] = int(eval(fp_params["gamma_only"]))
+                assert(fp_params["gamma_only"] == 0 or fp_params["gamma_only"] == 1),\
+                        "'gamma_only' should be either 0 or 1."
+                if fp_params["gamma_only"]==1:
+                    forward_files += ["KPT"]
+                else:
+                    if 'k_points' in jdata.keys() :
+                        forward_files += ["KPT"]
+                    elif 'fp_kpt_file' in jdata.keys():
+                        forward_files += ["KPT"]
+                    else:
+                        raise RuntimeError("Cannot find any k-points information")
+            else:
+                forward_files += ["KPT"]
```

Comment:
In all cases `forward_files += ["KPT"]`, why not simply write `forward_files = ["INPUT", "STRU", "KPT"]` ?

The validity of the kpoint information has been checked by `make_fp_abacus_scf`

Reply:
Thanks for pointing this. Revised.